ptr/generate.go

```diff
@@ -24,12 +24,20 @@ func main() {
 	}
 }
 
-func generateFile(filename string, tmplName string, types ptr.Scalars) error {
-	f, err := os.Create(filename)
-	if err != nil {
-		return fmt.Errorf("failed to create %s file, %v", filename, err)
+func generateFile(filename string, tmplName string, types ptr.Scalars) (err error) {
+	f, cerr := os.Create(filename)
+	if cerr != nil {
+		return fmt.Errorf("failed to create %s file, %v", filename, cerr)
 	}
-	defer f.Close()
+
```

Comment:
nit the `cerr` alternative error variable name isn't  be needed. As long as 1 item in a tuple is a new variable the `:=` can be used. Otherwise if all variables already exist in the block, `=` must be used, (or new var names)

https://play.golang.org/p/jkoWnscO0tt